internal/system/nvdevices/devices_test.go

```diff
@@ -126,9 +160,12 @@ func TestCreateControlDevices(t *testing.T) {
 			d.mknoder = mknode
 
 			err := d.CreateNVIDIAControlDevices()
-			require.ErrorIs(t, err, tc.expectedError)
+			if tc.hasError {
```

Comment:
`require.ErrorIs` should hanldle the case where `tc.expectedError` is `nil`. The need to do this may have been affected by us not wrapping an error correctly somewhere in the callchain.